### ISSUE TITLE
Custom mod types `mergeMods` setting is overruled by game-wide setting

### ISSUE THREAD

**Issue by agc93** [Nexus-Mods/Vortex#6268]
**Describe the bug**
Installing two mods with a custom mod type (manually set or matching `test` callback) where `mergeMods` is set to `false`:

```ts
context.registerModType('bs-map', 100, gameId => gameId === GAME_ID, getMapPath, testMap, { mergeMods: false });
```

will result in Vortex complaining about file conflicts, even though the "conflicting" files will be in completely different folders (thanks to `mergeMods: false`).

From what I can tell, this is caused by the current *game* having `mergeMods` set to true, even if the specific type sets it to not merge.

Is this expected? Is there a way to get this pattern, other than an installer that does its own "unmerging": add `${modId}/` to the start of the `destination` instructions?

**To Reproduce**
1. Register a game with `mergeMods` set to `true`
2. Register a mod type with `mergeMods` set to `false`
3. Install two mods using the custom type with the same filename in the contents
4. Vortex will complain about an unsolved file conflict even though they're deployed to two different folders.

**Expected behavior**
Vortex should see that their deployment paths are different (thanks to `mergeMods: false`) and not show conflicts

**Platform (please complete the following information):**
 - OS: Windows 10
 - Vortex Version 1.2.11

**Comment by TanninOne** [comment]
This is already fixed for the upcoming 1.2.12 release.
The modtype specific mergeMods flag was only considered for the actual deployment process, for the purpose of checking mod conflicts it only looked at the attribute set in the game.

**Comment by agc93** [comment]
Awesome! Thanks as always for the super-fast response @TanninOne 😄
